llvm/lib/Target/AMDGPU/GCNSubtarget.cpp

```diff
@@ -403,9 +405,13 @@ unsigned GCNSubtarget::getReservedNumSGPRs(const Function &F) const {
 std::pair<unsigned, unsigned>
 GCNSubtarget::computeOccupancy(const Function &F, unsigned LDSSize,
                                unsigned NumSGPRs, unsigned NumVGPRs) const {
+  bool IsDynamicVGPR = isDynamicVGPREnabled();
+  if (F.hasFnAttribute("amdgpu-dynamic-vgpr"))
+    IsDynamicVGPR = F.getFnAttribute("amdgpu-dynamic-vgpr").getValueAsBool();
```

Comment:
I think this needs to be consistent in all places thus a helper would be definitely needed.